doc/cookbook/splitting.md

```diff
@@ -66,3 +66,146 @@ $ pachctl put-file users master -c -f user_data.json --split json --target-file-
+$ pachctl get-file bar master /users/0000000000000000
+4,alice,aaa@place.com
+# Now we take the CSV header and apply it to the directory:
+$ cat users.csv | head -n 1 | pachctl put-header bar master users 
+# Now when we read an individual file, we see the header plus the contents
+$ pachctl get-file bar master /users/0000000000000000
+id,name,email
+4,alice,aaa@place.com
+$ pachctl get-file bar master /users
+id,name,email
+$ pachctl get-file bar master /users/**
+id,name,email
+4,alice,aaa@place.com
+7,bob,bbb@place.com
+```
+
+For more info see `pachctl put-header --help`
+
+## PG Dump / SQL Support
+
+You can also ingest data from postgres using split file.
+
+1) Generate your PG Dump file
+
+```
+$ pg_dump -t users -f users.pgdump
+$ cat users.pgdump 
+--
+-- PostgreSQL database dump
+--
+
+-- Dumped from database version 9.5.12
+-- Dumped by pg_dump version 9.5.12
+
+SET statement_timeout = 0;
+SET lock_timeout = 0;
+SET client_encoding = 'UTF8';
+SET standard_conforming_strings = on;
+SELECT pg_catalog.set_config('search_path', '', false);
+SET check_function_bodies = false;
+SET client_min_messages = warning;
+SET row_security = off;
+
+SET default_tablespace = '';
+
+SET default_with_oids = false;
+
+--
+-- Name: users; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE public.users (
+    id integer NOT NULL,
+    name text NOT NULL,
+    saying text NOT NULL
+);
+
+
+ALTER TABLE public.users OWNER TO postgres;
+
+--
+-- Data for Name: users; Type: TABLE DATA; Schema: public; Owner: postgres
+--
+
+COPY public.users (id, name, saying) FROM stdin;
+0	wile E Coyote	...
+1	road runner	\\.
+\.
+
+
+--
+-- PostgreSQL database dump complete
+--
+```
+
+
+2) Ingest using split file
+
+When you use `pachctl put-file --split ...` your pg dump file is split into
+three parts - the header, rows, and the footer. The header contains all the SQL
+statements in the pg dump that setup the schema and tables. The rows are split
+into individual files (or if you specify the `--target-file-datums` or 
+`--target-file-bytes` multiple rows per file). The footer contains the remaining
+SQL statements for setting up the tables.
+
+The header and footer are stored on the directory containing the rows. This way,
+if you request a `get-file` on the directory, you'll get just the header and
+footer. If you request an individual file, you'll see the header plus the row(s)
+plus the footer. If you request all the files with a glob pattern, e.g.
+`/directoryname/*`, you'll receive the header plus all the rows plus the footer,
+recreating the full pg dump. In this way, you can construct full or partial 
+pg dump files so that you can load full or partial data sets.
+
+```
+$ pachctl put-file data master -c -f users.pgdump --split sql
+$ pachctl put-file data master users --split sql -f users.pgdump 
+$ pachctl list-file data master
+NAME         TYPE SIZE 
+users        dir  914B 
+$ pachctl list-file data master /users
```

Comment:
So this went from too much cluttered info with all the SQL headers/footers to too little. Don't assume someone is going to read the CSV splitting docs above and then apply that logic to the sql splitting docs section. These sections should basically be standalone and walk through the same steps:

1. using --split
2. get-file without header
3. add header (or maybe its automatic when you use `--split sql`? It's unclear to me so that indicates the docs are missing something)
4. walk through example. I get that we dont want to print the whole PG header 4 times because that clutters, but you can cull that smartly and still have a complete step-by-step walkthrough.